dlt/common/destination/reference.py

```diff
@@ -185,6 +196,13 @@ def __init__(self, schema: Schema, config: DestinationClientConfiguration) -> No
         self.schema = schema
         self.config = config
 
+    @property
+    def dataset_name(self) -> str:
```

Comment:
this belongs to `DestinationClientDwhConfiguration` at best and will also cover the property there. also check `normalize_dataset_name` in the config above if you need an effective dataset name as used by sql code